src/features/tokens/useTokenBalance.tsx

```diff
@@ -46,11 +48,18 @@ export function getCachedTokenBalance(
     | undefined;
 }
 
-async function fetchTokenBalance(chainId: number, tokenAddress: Address, accountAddress: Address) {
-  logger.debug(
-    `Fetching balance for account ${accountAddress} token ${tokenAddress} on chain ${chainId}`,
-  );
-  const erc20 = getErc20Contract(tokenAddress, getProvider(chainId));
-  const balance = await erc20.balanceOf(accountAddress);
-  return balance.toString();
+async function fetchTokenBalance(chainId: ChainId, tokenAddress: Address, accountAddress: Address) {
+  if (isNativeToken(tokenAddress)) {
+    logger.debug(`Fetching balance for account ${accountAddress} native token on chain ${chainId}`);
```

Comment:
nit: if a MP is available it would be more readable to log the chain name

Reply:
this is just a log statement, not user facing